app/controllers/devise/devise_authy_controller.rb

```diff
@@ -121,25 +121,17 @@ def GET_verify_authy_installation
   end
 
   def POST_verify_authy_installation
-    token = Authy::API.verify({
-      :id => self.resource.authy_id,
-      :token => params[:token],
-      :force => true
-    })
+    token_valid = registration_token_valid?(resource.mfa_config)
 
-    self.resource.authy_enabled = token.ok?
+    resource.authy_enabled = token_valid
 
-    if token.ok? && self.resource.save
-      remember_device(@resource.id) if params[:remember_device].to_i == 1
+    if token_valid && resource.save
+      remember_device(resource.id) if params[:remember_device].to_i == 1
       record_authy_authentication
       set_flash_message(:notice, :enabled)
       redirect_to after_authy_verified_path_for(resource)
     else
-      if resource_class.authy_enable_qr_code
-        response = Authy::API.request_qr_code(id: resource.authy_id)
-        @authy_qr_code = response.qr_code
-      end
-      handle_invalid_token :verify_authy_installation, :not_enabled
+      handle_invalid_token :verify_authy_installation, :invalid_token
```

Comment:
need to rename from authy

Reply:
`:verify_authy_installation` is actually the name of a view for the `:verify_authy_installation` action. I think we should wait to change this when we rename that particular action and path.